repositories/slot_repository.go

```diff
@@ -0,0 +1,24 @@
+package repositories
+
+import (
+	"airport-app-backend/models"
+)
+
+type ISlotRepository interface {
+	RetrieveAllSlots(int, bool) ([]models.Slot, error)
+}
+
+func (sr *ServiceRepository) RetrieveAllSlots(page int, status bool) ([]models.Slot, error) {
+	var slots []models.Slot
+
+	query := sr.db.Offset(page * DEFAULT_PAGE_SIZE).Limit(DEFAULT_PAGE_SIZE)
+	if !status {
```

Comment:
simplify this by using `status` to generate an array of possible values, which in turn can be used in a `where` clause.